pkg/cmd/cockroach-sql/main.go

```diff
@@ -102,14 +152,45 @@ var sqlCmd = &cobra.Command{
 	// details and hints, which cobra does not do for us. Instead
 	// we do the printing in Main().
 	SilenceErrors: true,
+	// Prevent cobra from auto-generating a completions command,
+	// since we provide our own.
+	CompletionOptions: cobra.CompletionOptions{
+		DisableDefaultCmd: true,
+	},
+	// Support --version automatically.
+	Version: func() string {
+		info := build.GetInfo()
+		return "details:\n" + info.Long()
+	}(),
 }
 
-func runSQL(_ *cobra.Command, args []string) (resErr error) {
-	if url == "" && len(args) > 0 {
-		url = args[0]
+func runSQL(cmd *cobra.Command, args []string) (resErr error) {
+	fs := clienturl.FlagSetForCmd(cmd)
+	urlFlag := fs.Lookup(cliflags.URL.Name)
+
+	if len(args) > 0 {
+		if urlFlag.Changed {
+			return errors.New("cannot pass both --url and positional URL")
+		}
+		if err := urlFlag.Value.Set(args[0]); err != nil {
+			return err
+		}
+		urlFlag.Changed = true
 	}
-	if url == "" {
-		return errors.New("no connection URL specified")
+
+	// Translate the connection arguments to a connection URL.
+	connURL, err := clientconnurl.MakeClientConnURL(copts)
+	if err != nil {
+		return err
+	}
+
+	// How we're going to authenticate.
```

Comment:
nit: comment sounds like a question - maybe "Get how we're going to authenticate"